src/lib/hooks/useBlockNumber.tsx

```diff
@@ -81,8 +107,9 @@ export function BlockNumberProvider({ children }: { children: ReactNode }) {
           setChainBlock({ chainId: activeChainId, block: update })
         }
       },
+      mainnetValue: chainId === ChainId.MAINNET ? block : mainnetBlock,
```

Comment:
Why wouldn't this be up-to-date?

```suggestion
      mainnetValue: mainnetBlock,
```